@here/harp-features-datasource/lib/Features.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { FeatureGeometry, LineString, MultiPolygon, Point } from "@here/harp-datasource-protocol";
+import * as THREE from "three";
+
+/**
+ * Base class to create features.
+ */
+export abstract class MapViewFeature {
+    /**
+     * The type of the feature. The extended class should initialize this value. It defaults to
+     * "Point" in order to avoid allowing `null` or `undefined`.
+     */
+    type: FeatureGeometry["type"] = "Point";
+
+    /**
+     * A string identifying this feature.
+     */
+    uuid: string = THREE.Math.generateUUID();
```

Comment:
Is that `uuid` really needed?
I see it's only used for searching in current list of features.

Wouldb't matching be simpler when just comparing identity of objects?